### PR TITLE
Improved app logger

### PR DESCRIPTION
**Pull request by ravenjohn**
What's changed:
  * created `helpers/logger.js`
  * replaced `winston`'s default console logger with a console logger that includes timestamp
  * added a file logger for warn or error logs
  * handled daily access logs
  * removed the unmaintained module `file-stream-rotator`, used `winston-daily-rotate-file` instead

image

### CODE REVIEW

File: helpers/logger.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    level: 'warn',
+    json: false
+});
+
+
+// handles access logs
+module.exports = new (winston.Logger)({
+    transports: [
+        new daily({
+            formatter: a => a.message.trim(),
```
Comment (pandafox) [inline helpers/logger.js]: Don't we need to check 'a.messge' to make sure it has value?

File: helpers/logger.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+const moment  = require('moment');
+
+const stamp   = () => moment.utc().format('YYYY-MM-DD HH:mm:ss:SSS[ms]');
+
+
+// remove default
+winston.remove(winston.transports.Console);
+
+// replace with a better one
+winston.add(winston.transports.Console, {
```
Comment (pandafox) [inline helpers/logger.js]: I think this comment would make more sense if combined w previous comment, and put the remove/add code together

File: helpers/logger.js (hunk 1 above)
Comment (ravenjohn) [inline helpers/logger.js]: we don't need to. morgan is supplying it for us.

like this
```
::ffff:127.0.0.1 - - [08/Sep/2016:01:47:51 +0000] "GET / HTTP/1.1" 404 33 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/53.0.2785.89 Safari/537.36"
```

File: helpers/logger.js (hunk 2 above)
Comment (ravenjohn) [inline helpers/logger.js]: oke

**Comment by jekku** [comment]
awww yisss timestamp

**Pull request merged**